src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,40 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final Map<EventListener, ExecutorService> listenerExecutors = new ConcurrentHashMap<>();
 
+    public void registerListener(EventListener listener) {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        listenerExecutors.put(listener, executorService);
```

Comment:
sorry, but I cannot get idea why we need separate executor service for each listener